ACPs/224-dynamic-gas-limit-in-subnet-evm/README.md

```diff
@@ -0,0 +1,233 @@
+
+```json
+{
+  ...
+  "acp224Timestamp": ...,
+  "acp224FeeConfig": {
+    "targetGas": ...,
+    "minGasPrice": ...,
+    "maxCapacityFactor": ...,
+    "timeToDouble": ...,
+    "targetBlockRate": ...,
+    "minBlockGasCost": ...,
+    "maxBlockGasCost": ...,
+    "blockGasCostStep": ...
+  }
+}
+
+```
+
+### New ACP176 Fee Manager Precompile
+
+A new fee manager precompile will be required to dynamically changing the parameters. The precompile will be similar to the existing fee manager precompile. The solidity interface will be as follows:
+
+```solidity
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+import "./IAllowList.sol";
+
+interface IACP176FeeManager is IAllowList {
+  struct ACP176FeeConfig {
+    uint256 targetGas;
+    uint256 minGasPrice;
+    uint256 maxCapacityFactor;
+    uint256 timeToDouble;
+    uint256 targetBlockRate;
+    uint256 minBlockGasCost;
+    uint256 maxBlockGasCost;
+    uint256 blockGasCostStep;
+  }
+  event FeeConfigChanged(address indexed sender, ACP176FeeConfig oldFeeConfig, ACP176FeeConfig newFeeConfig);
+
+  // Set fee config fields to contract storage
+  function setFeeConfig(
+    uint256 targetGas;
+    uint256 minGasPrice;
+    uint256 maxCapacityFactor;
+    uint256 timeToDouble;
+    uint256 targetBlockRate;
+    uint256 minBlockGasCost;
+    uint256 maxBlockGasCost;
+    uint256 blockGasCostStep;
+  ) external;
+
+  // Get fee config from the contract storage
+  function getFeeConfig()
+    external
+    view
+    returns (
+      uint256 targetGas;
+      uint256 minGasPrice;
+      uint256 maxCapacityFactor;
+      uint256 timeToDouble;
+      uint256 targetBlockRate;
+      uint256 minBlockGasCost;
+      uint256 maxBlockGasCost;
+      uint256 blockGasCostStep;
+    );
+
+  // Get the last block number changed the fee config from the contract storage
+  function getFeeConfigLastChangedAt() external view returns (uint256 blockNumber);
+}
+```
+
+The precompile will have a `targetGas` parameter that is used to set the target gas consumed per second for ACP-176. This will override the `gas-target` in node config. If the new fee manager precompile is not activated, validators can keep using `gas-target` in node config to set the target gas consumed per second (as in C-Chain).
+
+```mermaid
+flowchart TD
+    A[ACP-224 activated] --> B{Is ACP176FeeManager precompile active?}
+
+    B -- Yes --> C[Use targetGas from precompile storage]
+
+    B -- No --> D{Is gas-target set in node config?}
+    D -- Yes --> E[Use gas-target from node config]
+
+    D -- No --> F{Is targetGas set in chainConfig.acp176FeeConfig?}
+    F -- Yes --> G[Use targetGas from chain config]
+
+    F -- No --> H[Use default targetGas e.g. 2 million]
```

Comment:
For ACP-176, the implementation actually uses the current on-chain value as the "default" effectively abstaining from having a preference if none is set. I think we should do the same here.

Reply:
right, I was under impression that we were using a config default.